ambry-mysql/src/main/java/com/github/ambry/repair/MysqlRepairRequestsDb.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+package com.github.ambry.repair;
+
+import com.github.ambry.utils.Utils;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.repair.RepairRequestRecord.OperationType;
+import static com.github.ambry.utils.Utils.*;
+
+
+/**
+ * MysqlRepairRequestsDb, using mysql db to implement the RepairRequestsDb
+ * The class to manipulate the AmbryRepairRequests DB.
+ * The AmbryRepairRequests DB has the records of {@link RepairRequestRecord}.
+ * These are partially failed requests which need to get fixed.
+ */
+public class MysqlRepairRequestsDb implements RepairRequestsDb {
+  private static final Logger logger = LoggerFactory.getLogger(MysqlRepairRequestsDb.class);
+
+  // table name
+  public static final String REPAIR_REQUESTS_TABLE = "ambry_repair_requests";
+
+  // column names
+  private static final String BLOB_ID = "blobId";
+  private static final String PARTITION_ID = "partitionId";
+  private static final String SOURCE_HOST_NAME = "sourceHostName";
+  private static final String SOURCE_HOST_PORT = "sourceHostPort";
+  private static final String OPERATION_TYPE = "operationType";
+  private static final String OPERATION_TIME = "operationTime";
+  private static final String LIFE_VERSION = "lifeVersion";
+  private static final String EXPIRATION_TYPE = "expirationTime";
+
+  // LOCAL_CONSISTENCY_TODO continue the Get from one token
+  // LOCAL_CONSISTENCY_TODO set the max ResultSet count
+  /**
+   * Select the records for one partition with the oldest operation time.
+   */
+  // @formatter:off
+  private static final String GET_QUERY = String.format(""
+      + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+      + "FROM %s "
+      + "WHERE %s = ? "
+      + "ORDER BY %s ASC "
+      + "LIMIT 100",
```

Comment:
shouldn't we use the listMaxResults from configuration here.

Reply:
Added  the limit config